kiosk/kiosk_browser/browser_widget.py

```diff
@@ -29,9 +32,38 @@ def __init__(self, url, *args, **kwargs):
         self.setSizePolicy(policy)
 
     def clean_and_load(self, url):
-        self.setPage(QWebEnginePage())
-        self.load(url)
+
+        # Recreate a new page in order to clear the screen the moment the
+        # shortcut is pressed
+        page = QWebEnginePage(self._profile)
```

Comment:
We’re recreating a page instance each time, instead of reusing the same one. I added a comment to indicate why we do this.

From [Qt documentation](https://doc.qt.io/qt-5/qwebengineview.html#setPage), I understand that `setPage` will free the current page when loading a new page:

>  If the current page is a child of the web view, it will be deleted.

I’ll profile the kiosk browser locally by changing the urls a lot following [this guide](https://medium.com/zendesk-engineering/hunting-for-memory-leaks-in-python-applications-6824d0518774)